The IcebergIO Unit Tests job is flaky

The IcebergIO Unit Tests is failing over 50% of the time.
Please visit https://github.com/apache/beam/actions/workflows/IO_Iceberg_Unit_Tests.yml?query=is%3Afailure+branch%3Amaster to see all failed workflow runs.
See also Grafana statistics: http://metrics.beam.apache.org/d/CTYdoxP4z/ga-post-commits-status?orgId=1&viewPanel=11&var-Workflow=IcebergIO%20Unit%20Tests

Failures look like this:
```
2026-07-09T18:37:41.5816304Z BeamRowWrapperTest > testNullFieldHandling FAILED
2026-07-09T18:37:41.5817394Z     java.lang.NoClassDefFoundError: Could not initialize class org.apache.beam.sdk.io.iceberg.BeamRowWrapperTest
2026-07-09T18:37:41.5818690Z         at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
2026-07-09T18:37:41.5819877Z         at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)
...
2026-07-09T18:37:41.5865609Z         Caused by:
2026-07-09T18:37:41.5866455Z         java.lang.ExceptionInInitializerError: Exception java.lang.AssertionError: micros_instant logical type encountered a Java Instant with greater than microsecond precision. [in thread "Test worker"]
2026-07-09T18:37:41.5867581Z             at org.apache.beam.sdk.schemas.logicaltypes.MicrosInstant.toBaseType(MicrosInstant.java:56)
2026-07-09T18:37:41.5868354Z             at org.apache.beam.sdk.schemas.logicaltypes.MicrosInstant.toBaseType(MicrosInstant.java:40)
2026-07-09T18:37:41.5869115Z             at org.apache.beam.sdk.values.RowUtils$CapturingRowCases.processLogicalType(RowUtils.java:547)
2026-07-09T18:37:41.5869890Z             at org.apache.beam.sdk.values.RowUtils$RowFieldMatcher.match(RowUtils.java:167)
2026-07-09T18:37:41.5870568Z             at org.apache.beam.sdk.values.RowUtils$CapturingRowCases.processRow(RowUtils.java:420)
2026-07-09T18:37:41.5871439Z             at org.apache.beam.sdk.values.RowUtils$RowFieldMatcher.match(RowUtils.java:163)
2026-07-09T18:37:41.5872144Z             at org.apache.beam.sdk.values.Row$Builder.build(Row.java:870)
2026-07-09T18:37:41.5872782Z             at org.apache.beam.sdk.io.iceberg.BeamRowWrapperTest.<clinit>(BeamRowWrapperTest.java:109)

```
Looks like [failures](https://github.com/apache/beam/actions/workflows/IO_Iceberg_Unit_Tests.yml?query=event%3Aschedule) started after bumping CI to Java21: https://github.com/apache/beam/commit/123dc7feaa45fc4b6ca96788759eec18c364c462

I don't see any changes to the Iceberg module around the time failures started

--- a/sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/BeamRowWrapperTest.java
+++ b/sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/BeamRowWrapperTest.java
@@ -85,6 +85,8 @@ public class BeamRowWrapperTest {
                   Types.NestedField.required(1, "nested_int", Types.IntegerType.get()))),
           Types.NestedField.required(14, "pass_through_field", Types.IntegerType.get()));
   private static final UUID TEST_UUID = UUID.randomUUID();
+  private static final Instant TEST_MICROS_INSTANT =
+      Instant.ofEpochSecond(1_609_459_200L, 123_456_000);
   private static final Row NESTED_ROW = Row.withSchema(NESTED_BEAM_SCHEMA).addValue(999).build();
   private static final Row ROW =
       Row.withSchema(BEAM_SCHEMA)
@@ -99,7 +101,7 @@ public class BeamRowWrapperTest {
                   .array(),
               new BigDecimal("123.45"),
               org.joda.time.Instant.now(),
-              Instant.now(),
+              TEST_MICROS_INSTANT,
               LocalDateTime.now(ZoneId.systemDefault()),
               LocalDate.now(ZoneId.systemDefault()),
               LocalTime.now(ZoneId.systemDefault()),
@@ -180,10 +182,7 @@ public class BeamRowWrapperTest {
 
   @Test
   public void testMicrosInstantLogicalTypeConversion() {
-    Instant javaInstant = ROW.getLogicalTypeValue(7, Instant.class);
-    long expectedMicrosInstant =
-        TimeUnit.SECONDS.toMicros(javaInstant.getEpochSecond()) + javaInstant.getNano() / 1000;
-    assertEquals(expectedMicrosInstant, (long) WRAPPER.get(7, Long.class));
+    assertEquals(1_609_459_200_123_456L, (long) WRAPPER.get(7, Long.class));
   }
 
   @Test

